appinfo/app.php

```diff
@@ -22,17 +22,13 @@
  *
  */
 
-OCP\Util::addStyle('files_3d', 'style');
-OCP\Util::addScript('files_3d', 'vendor/three.min');
-OCP\Util::addScript('files_3d', 'vendor/controls/OrbitControls');
-OCP\Util::addScript('files_3d', 'vendor/libs/inflate.min');
-OCP\Util::addScript('files_3d', 'vendor/loaders/LoaderSupport');
-OCP\Util::addScript('files_3d', 'vendor/loaders/ColladaLoader');
-OCP\Util::addScript('files_3d', 'vendor/loaders/FBXLoader');
-OCP\Util::addScript('files_3d', 'vendor/loaders/GLTFLoader');
-OCP\Util::addScript('files_3d', 'vendor/loaders/OBJLoader2');
-OCP\Util::addScript('files_3d', 'vendor/loaders/MTLLoader');
-OCP\Util::addScript('files_3d', 'loader');
+ $eventDispatcher = \OC::$server->getEventDispatcher();
+ $eventDispatcher->addListener(
+ 	'OCA\Files::loadAdditionalScripts',
```

Comment:
Should be LoadViewer 
https://github.com/nextcloud/viewer#add-the-viewer-to-your-app

Reply:
I'm a bit confused. I don't have (and don't need, I think?) any controllers. I tried to copy the appinfo setup from the viewer app itself, but that doesn't work. Any ideas :blush: ?

Reply:
Yes, because you don't have a dedicated page view :)
The mechanism is the same, 

You just need to listen to the Viewer event, the files app is firing it
`use OCA\Viewer\Event\LoadViewer;`

https://github.com/nextcloud/viewer/blob/master/lib/Listener/LoadViewerScript.php
https://github.com/nextcloud/viewer/blob/b16ba3ab0dc81a9e98a88a9917936c527e94cca6/lib/AppInfo/Application.php#L42

Reply:
Cleanup your app.php so it query your main Application.php
https://github.com/nextcloud/viewer/blob/master/appinfo/app.php

Reply:
I copied the (imo 😅 ) interesting parts from app.php and Application.php from viewer to my app, but that didn't work. I'll push my latest changes tomorrow :)

Reply:
Sure, pûsh and i'll have a look :)

Reply:
Pushed my changes :)

Reply:
Commented!